File: python/structures/src/sorts/bubble.py
from structures.src.util.sort_util import is_sorted, swap

def sort(arr):
    # want this to be pure
    out = list(arr)

    # keep doing it until we're sorted
    while not is_sorted(out):
        for i in range(0, len(out) - 1):
            # if this is out of order, swap it to the next slot
            if out[i] > out[i + 1]:
                swap(out, i, i + 1)

    return out

File: c/sorts/bubble.c
#include "bubble.h"

#include "../util/swap.h"
#include "issorted.h"

void Bubble_sort(int *arr, int length) {
    while(issorted(arr, length) != 1) {
        for(int i = 0; i < length - 1; i++) {
            if(arr[i] > arr[i + 1]) {
                swap(arr, i, i + 1);
            }
        }
    }
}
